futures-util/src/stream/select.rs

```diff
@@ -1,50 +1,67 @@
 use super::assert_stream;
-use crate::stream::{Fuse, StreamExt};
+use crate::stream::{select_with_strategy, PollNext, SelectWithStrategy};
 use core::pin::Pin;
 use futures_core::stream::{FusedStream, Stream};
 use futures_core::task::{Context, Poll};
 use pin_project_lite::pin_project;
 
 pin_project! {
     /// Stream for the [`select()`] function.
-    #[derive(Debug)]
```

Comment:
Why Debug impl is removed?

Reply:
ah, sorry. My bad. I intended to put it back but forgot. The reason is this:

```
error: implementation of `Debug` is not general enough
  --> futures-util/src/stream/select.rs:10:14
   |
10 |     #[derive(Debug)]
   |              ^^^^^ implementation of `Debug` is not general enough
   |
   = note: `Debug` would have to be implemented for the type `for<'r> fn(&'r mut PollNext) -> PollNext`
   = note: ...but `Debug` is actually implemented for the type `fn(&'0 mut PollNext) -> PollNext`, for some specific lifetime `'0`
```

I think we need a manual implementation, but I'm not sure what to put in it. I suppose the derive only implemented it when the generics implemented it. We can access inner through SelectWithStrategy to call their Debug I suppose.

Reply:
We can do this:
```rs
impl<St1, St2> fmt::Debug for Select<St1, St2>
where
    St1: fmt::Debug,
    St2: fmt::Debug,
{
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!( f, "Select over {:?}", self.inner.get_ref() )
    }
}
```

With the two repeat streams of the example it gives:
```
Select over (Repeat { item: 1 }, Repeat { item: 2 })
```
I should probably also add a Debug to `SelectWithStrategy`. What do you think of the above implementation?

Reply:
Ah, I think we need to change the `Debug` impl of `SelectWithStrategy` not to print closure (like https://github.com/rust-lang/futures-rs/pull/1677 did). After that, we could re-add `#[derive(Debug)]` to `Select`.

Reply:
Ok, makes sense. The result for `Select` is:

`Select { inner: SelectWithStrategy { stream1: Fuse { stream: Repeat { item: 1 }, done: false }, stream2: Fuse { stream: Repeat { item: 2 }, done: false } } }`